galley/pkg/api/distrib/watch.proto

```diff
@@ -0,0 +1,285 @@
+//    state == DOES_NOT_EXIST. At least one message for element="" will
+//    be delivered.
+//
+// 2. `resume_marker` == "now": there will be exactly one message with
+//    element = "" and state INITIAL_STATE_SKIPPED. The client cannot
+//    assume whether or not the entity exists after receiving this
+//    message.
+//
+// 3. `resume_marker` has a value R from a preceding watch call on this
+//    entity: The same messages as described in (1) will be delivered to
+//    the client, except that any information implied by messages received
+//    on the preceding call up to and  including R may not be
+//    delivered. The expectation is that the client will start with state
+//    it had built up from the preceding watch call, apply the changes
+//    received from this call, and build an up-to-date view of the entity
+//    without having to fetch a potentially large amount of information
+//    that has not changed. Note that some information that had already
+//    been delivered by the preceding call might be delivered again.
+//
+// ## Ordering and Reliability
+//
+// The Change messages that apply to a particular element of the entity are
+// delivered eventually in order without loss for the duration of the RPC. Note
+// however that if multiple Changes apply to the same element, the
+// implementation is free to suppress them and deliver just the last one. The
+// underlying system must provide the guarantee that any relevant update
+// received for an entity E after a client's watch point for E MUST be delivered
+// to that client.
+//
+// These tight guarantees allow for the following simplifications in the client:
+//
+//   1. The client does not need to have a separate polling loop to make up for
+//      missed updates.
+//
+//   2. The client does not need to manage timestamps/versions manually; the
+//      last update delivered corresponds to the eventual state of the entity.
+//
+// Example: a calendar entry may have elements named { "starttime", "endtime",
+// "attendees" } with corresponding values or it may have a single element name
+// "entry" with a serialized proto for the calendar entry.
+//
+// ## Ordering constraints for parents/descendants
+//
+// The Watch API provides guarantees regarding the order in which
+// messages for a parent and its descendants are delivered:
+//
+// 1. The creation of an ancestor (i.e., the first EXISTS message for
+//    the ancestor) is reported before the creation of any of its
+//    descendants.
+//
+// 2. The deletion of an ancestor (via a DOES_NOT_EXIST message)
+//    implies the deletion of all its descendants. The service will
+//    not deliver any messages for the descendants until the parent
+//    has been recreated.
+
+
+// The service that a client uses to connect to the watcher system.
+// The errors returned by the service are in the canonical error space,
+// see [google.rpc.Code][].
+service Watcher {
```

Comment:
We have:
- ADS - an interface for pushing config changes
- K8S - with its own interface for pushing config changes

Can you explain why we need a 3rd one, with different name, semantics, etc ? 
Nothing can be reused from the 2 interfaces we already use ?